runtime/autoload/dist/ft.vim

```diff
@@ -644,7 +644,7 @@ export def FTpl()
     # recognize Prolog by specific text in the first non-empty line
     # require a blank after the '%' because Perl uses "%list" and "%translate"
     var line = getline(nextnonblank(1))
-    if line =~ '\<prolog\>' || line =~ '^\s*\(%\+\(\s\|$\)\|/\*\)' || line =~ ':-'
+    if line =~ '\<prolog\>' || line =~ '^\(:-\|%\|\/\*\)\|\.$'
```

Comment:
This doesn't match the comment above that requires the `%\+\(\s\|$\)` pattern:
```
    # require a blank after the '%' because Perl uses "%list" and "%translate"
```
Meanwhile, the comment in `ProtoCheck()` needs to be updated, as it no longer matches the code and that function has nothing to do with Perl.

Reply:
@zeertzjq Do you suggest removing that line (645)? Or should it be better to write there a comment describing the RegEx?

Reply:
No, I think the requirement in the comment needs to be respected.

Reply:
Aaaaaah I think I get it.

A Perl file may begin with, for example:

```perl
%data = ('John Paul', 45, 'Lisa', 30, 'Kumar', 40);
```

And it may get confused for a Prolog comment.

Did I understand correctly?

Reply:
Yes

Reply:
Alright, thanks for the patience, I modified the Regex to this:

```vim
let regex = '^\(:-\|%\+\(\s\|$\)\|\/\*\)\|\.$'

let pattern1 = ':-module(test/1,'
let pattern2 = '% comment'
let pattern3 = '/* multiline comment'
let pattern4 = 'rule(test, 1.7).'
let patternX = 'a'
let patternY = '%data = ("John Paul", 45, "Lisa", 30, "Kumar", 40);'

if pattern1 =~ regex
  echo 'OK'
else
  echo 'ERROR'
endif

if pattern2 =~ regex
  echo 'OK'
else
  echo 'ERROR'
endif

if pattern3 =~ regex
  echo 'OK'
else
  echo 'ERROR'
endif

if pattern4 =~ regex
  echo 'OK'
else
  echo 'ERROR'
endif

if patternX =~ regex
  echo 'ERROR'
else
  echo 'OK'
endif

if patternY =~ regex
  echo 'ERROR'
else
  echo 'OK'
endif
```

All tests are passing.

The `patternX` and `patternY` aren't supposed to match (I'm adding them as tests too).

Do you happen to see any other weird thing?

Reply:
There may be leading whitespace before the text, so I think it should still be `^\s*` at the start. That may apply to `ProtoCheck()` as well.

Reply:
Alright, I improved the Regex for leading and trailing whitespace.

I'm not that experienced with PRs. I did `git add .` and `git commit`.
Now I should do `git push fork update-filetype-FTpl`, right?

Reply:
Yes

Reply:
Thanks!